include/circt/Dialect/RTL/RTLVisitors.h

```diff
@@ -79,9 +79,10 @@ class StmtVisitor {
   ResultType dispatchStmtVisitor(Operation *op, ExtraArgs... args) {
     auto *thisCast = static_cast<ConcreteType *>(this);
     return TypeSwitch<Operation *, ResultType>(op)
-        .template Case<OutputOp, InstanceOp>([&](auto expr) -> ResultType {
-          return thisCast->visitStmt(expr, args...);
-        })
+        .template Case<OutputOp, InstanceOp, TypedeclOp>(
```

Comment:
TypedeclOp can only exist in TypedeclScopeOp, so they aren't statements.  Please don't add them to this visitor.  If we eventually need a visitor, it should be specific to entities that can only exist in `TypedeclScopeOp`, perhaps named `TypedeclScopeVisitor`.

Reply:
Since these can exist as top-level entities, like modules, I've added a separate top-level emitter for them, based on a new visitor that just handles the children of TypeScopeOp.